File: algo/9. Palindrome Number.py
import math

class Solution:
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x<0:
            return False
        
        if x==0:
            return True
        
        length = int(math.log10(x)) + 1
        
        while length>1:
            last = x%10
            #print('last: {}'.format(last))
            first = x//(10**(length-1))
            #print('first: {}'.format(first))
            if last != first:
                return False
            x = (x-first*(10**(length-1))-last)/10
            #print(x)
            length -= 2
            
        return True

File: round3/9. Palindrome Number.cs
public class Solution {
    public bool IsPalindrome(int x) {
        if (x < 0) return false;
        
        int b = 1;
        
        while (x / b >= 10) b *= 10;
        
        while (x > 0) {
            int left = x / b;
            int right = x % 10;
            if (left != right) return false;
            x = (x - left * b) / 10;
            b = b / 100;
        }
        
        return true;
    }
}